### ISSUE TITLE
antsMotionCorrStats framewise vs. MOCOparams.csv

### ISSUE THREAD

**Issue by johann997** [ANTsX/ANTs#1169]
This may seem like an amateur question, but I am finding myself to be quite stuck. 
I am playing around with the minimal fMRI pre-processing pipeline [here](https://stnava.github.io/fMRIANTs/) (instead of using FSL’s MCFLIRT), as we are wanting to implement a full ANTs pipeline. 

Looking at MOCOparams.csv I cannot work out;
 	1.1 Which parameters correspond to each motion (i.e. translation, rotation, scaling and shearing)?
	1.2 What units the columns are in (pixels vs. mm)?
	1.3 What is the MetricPost?

I have tried to read the ITK docs [ITK docs](https://itk.org/Doxygen/html/classitk_1_1AffineTransform.html) suggested in  #1094 however, I am still stuck. 

In short, I am wanting to work out the frame wise displacement after motion correction and have run antsMotionCorrStats instead of using the .csv directly
`antsMotionCorrStats -m MOCOparams.csv -x fmri.nii.gz -o stats.nii.gz -d`

However, I am still unsure:
        2.1 What the units are?
        2.2  For the mask, is it better to use the original uncorrected fMRI image or the motion corrected image?

Any advice or direction would be greatly appreciated.

**Comment by gdevenyi** [comment]
https://github.com/ANTsX/ANTs/issues/1094

**Comment by johann997** [comment]
> #1094

Unfortunately, I had already come across #1094 and couldn't make sense of the ITK documentation.

**Comment by gdevenyi** [comment]
The 12 parameters correspond to the entries in an Affine transformation matrix https://www.brainvoyager.com/bv/doc/UsersGuide/CoordsAndTransforms/SpatialTransformationMatrices.html

The order of the entries is:
> This class provides several methods for setting the matrix and vector defining the transform. To support the registration framework, the transform parameters can also be set as an Array<double> of size (NDimension + 1) * NDimension using method SetParameters(). The first (NDimension x NDimension) parameters defines the matrix in row-major order (where the column index varies the fastest). The last NDimension parameters defines the translation in each dimensions.

So the first 9 entries correspond to the matrix in row-major order, and the last 3 correspond to the translation, these are defined in the ITK LPS coordinate system. The matrix entries have a complex relationship to the translation/rotation/scaling. In MINC land I could use `xfm2param` to decompose into the linear operations you're describing, I'm not sure what in ITK land will do so. The matrix entries should not be interpreted as having "units" but the LPS space is defined in units of mm.

The metric refers to (pre) the registration metric before optimization and (post) afterwards. Its exact value depends on which metric you choose.

Framewise displacement is probably mm. The mask should not be either of the things you suggest, but rather, a mask, defining with 1/0 where the region of interest is to compute the mean framewise displacement.

**Comment by johann997** [comment]
Ah ok, I highly appreciate the description. For some reason I was bent on assuming the columns were the different motion parameters. 
`xfmparam` is a nice note, I had a look for some function that could format a Warp.nii to .xfm but the best I could find was `nii2mnc`, [(for more info)](http://bic-mni.github.io/man-pages/man/nii2mnc.html) but I will keep looking.

I should have stated that I would binarise the images before passing them as a mask. But yes, I suppose it doesn't need to be exact, as long as it cover the region of interest. 

Again, thanks for the information, this will get me going!

**Comment by gdevenyi** [comment]
>  I had a look for some function that could format a Warp.nii to .xfm but the best I could find was nii2mnc, (for more info) but I will keep looking.

Well actually, we're talking about affine transforms, so there's no nii file to begin with. The matrix entries in the csv file could be manually entered into a MINC affine xfm (some details at https://en.wikibooks.org/wiki/MINC/Tools/mni_autoreg). You may need to invert the xfm before decomposing because ANTs/MINC conceptualize the direction of transforms differently.
